tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/call_to_action.html

```diff
@@ -0,0 +1,9 @@
+{% load wagtailcore_tags %}
+
+<div class="call-to-action-block">
+    <p>{{ value.text }}</p>
+
+    {% if value.button_text and value.button_link %}
+        <a href="{{ value.button_link.get_link }}" class="btn">{{ value.button_text }}</a>
```

Comment:
**Issue**: this `href` does not work.

**Suggestion**

To fix this, you could consider

1. creating an `EmailLinkBlock`, similar to the `ExternalLinkBlock` (in place of the link_url, we'd have an email address)
2. refactoring the `LinkBlock` so that it now has a third field, say, `email_link = EmailLinkBlock`()`
3. updating the `button_link` field in `CallToActionBlock`, so that it now uses the updated `LinkBlock` above
4. updating the template accordingly

That's just one approach, feel free to use any other alternatives :smile: 

Please give me a shout if you need us to discuss it further :)

Reply:
Ah, thanks for flagging this. I've gone for a slightly different approach that it looks like you may have used on wagtail-kit so as not to alter LinkBlock() which needs just internal & external links for the QuoteBlock(). This all seems to be working well

Reply:
Nice one!